File: luci-app-atinout/luci-app-atinout/luasrc/model/cbi/modem/atconfig.lua
-- Copyright 2020 Rafał Wabik (IceG) - From eko.one.pl forum
-- Licensed to the GNU General Public License v3.0.

local util = require "luci.util"
local fs = require "nixio.fs"
local sys = require "luci.sys"
local http = require "luci.http"
local dispatcher = require "luci.dispatcher"
local http = require "luci.http"
local sys = require "luci.sys"
local uci = require "luci.model.uci".cursor()

local ATC_FILE_PATH = "/etc/atcommands.user"

local m
local s
local dev1
local try_devices1 = nixio.fs.glob("/dev/tty[A-Z][A-Z]*")

m = Map("atinout", translate("Atinout Configuration"),
	translate("Configuration panel for atinout."))

s = m:section(NamedSection, 'general' , "atinout" , "<p>&nbsp;</p>" .. translate("AT Commands Terminal Settings"))
s.anonymous = true

dev1 = s:option(Value, "atcport", translate("AT Command Sending Port"))
if try_devices1 then
local node
for node in try_devices1 do
dev1:value(node, node)
end
end

local atc = s:option(TextValue, "user_atcommands", translate("User AT Commands"), translate("Each line must have the following format: 'AT Command name;AT Command'. Save to file '/etc/atcommands.user'."))
atc.rows = 20
atc.rmempty = false

function atc.cfgvalue(self, section)
    return fs.readfile(ATC_FILE_PATH)
end

function atc.write(self, section, value)
    		value = value:gsub("\r\n", "\n")
    		fs.writefile(ATC_FILE_PATH, value)
end

return m

File: luci-app-atinout/htdocs/luci-static/resources/view/modem/atconfig.js
'use strict';
'require form';
'require fs';
'require view';
'require uci';
'require ui';
'require tools.widgets as widgets'

/*
	Copyright 2022-2023 Rafał Wabik - IceG - From eko.one.pl forum

	Modified for atinout by Konstantine Shevlakov <shevlakov@132lan.ru> 2023
	
	Licensed to the GNU General Public License v3.0.
*/


var cmddesc = _("Each line must have the following format: 'At command description;AT command'. For user convenience, the file is saved to the location <code>/etc/atcommands.user</code>."); 

return view.extend({
	load: function() {
		return fs.list('/dev').then(function(devs) {
			return devs.filter(function(dev) {
				return dev.name.match(/ttyUSB/) || dev.name.match(/ttyAC/);
			});
		});
	},

	render: function(devs) {
		var m, s, o;
		m = new form.Map('atinout', _('Configuration'), _('Configuration panel for atinout.'));

		s = m.section(form.TypedSection, 'atinout', '', _(''));
		s.anonymous = true;

		o = s.option(form.Value, 'atc_port', _('Port for communication with the modem'), 
			_("Select serial modem port."));
		devs.sort((a, b) => a.name > b.name);
		devs.forEach(dev => o.value('/dev/' + dev.name));
		
		o.placeholder = _('Please select a port');
		o.rmempty = false;

		o = s.option(form.TextValue, '_tmpl', _('User AT commands'), cmddesc);
		o.rows = 20;
		o.cfgvalue = function(section_id) {
			return fs.trimmed('/etc/atcommands.user');
		};
		o.write = function(section_id, formvalue) {
			return fs.write('/etc/atcommands.user', formvalue.trim().replace(/\r\n/g, '\n') + '\n');
		};

		return m.render();
	}
});
